Sources/SwiftParser/TokenPrecedence.swift

```diff
@@ -219,7 +219,7 @@ enum TokenPrecedence: Comparable {
       // Operator stuff
       .operator, .precedencegroup,
       // Misc
-      .import:
+      .import, .__consuming:
```

Comment:
You adding `__consuming` here actually got me thinking if we really need to specify a recovery precedence in `DeclarationModifier` (sorry for not noticing this earlier). I just checked and all keywords in there except for the following are exclusively used for declarations, so we should be able to just set their recovery precedence in `TokenPrecedence`.

- `weak` 
- `unowned`
- `isolated`
- `rethrows`
- `async`
- `reasync`
- `_const`
- `consuming`
- `borrowing`

Just for reference: What I did was right click on all of the `Keyword.something` references and select Find -> “Find Selected Symbol in Workspace” and then judging on where the keyword was used. If it was only used in `DeclarationModifier`, it’s unconditionally a declaration keyword.

Reply:
I added `__consuming` is because it's `isLexerClassifiedKeyword=true`. It seems assigned precedence to most of the modifier keywords there also make sense. Already done this.